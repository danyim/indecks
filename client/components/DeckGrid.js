import React from 'react';
import Deck from './Deck';
import AddDeck from './AddDeck';

const DeckGrid = React.createClass({
  render() {
    return (
      <div className="deck-grid">
        <AddDeck />
        {this.props.decks.map((deck, i) =>
          <Deck {...this.props} key={i} i={i} deck={deck} />
        )}
      </div>
    )
  }
});

export default DeckGrid;
