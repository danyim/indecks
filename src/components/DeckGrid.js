import React from 'react';
import Deck from './Deck';
import styles from '../styles/components/DeckGrid';

class DeckGrid extends React.Component {
  render() {
    const emptyMsg =
      this.props.decks.length === 0 ?
        <p>Click the + button on the top left to add a deck</p>
        : '';

    return (
      <section className={`${styles['deck-grid']} wrap-row`}>
        {this.props.decks.map((deck, i) =>
          <Deck {...this.props} key={i} i={i} deck={deck} />
        )}
        {emptyMsg}
      </section>
    )
  }
}

DeckGrid.propTypes = {};
DeckGrid.defaultProps = {};

export default DeckGrid;
