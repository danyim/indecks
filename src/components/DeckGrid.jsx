import React from 'react'
import PropTypes from 'prop-types'
import { TransitionMotion, spring } from 'react-motion'
import { DeckShape } from './__commonShapes'
import Deck from './Deck'
import Splash from './Splash'
import styles from '../styles/components/DeckGrid.styl'

class DeckGrid extends React.Component {
  static propTypes = {
    push: PropTypes.func.isRequired,
    decks: PropTypes.arrayOf(DeckShape).isRequired,
    removeDeck: PropTypes.func.isRequired,
    loadSampleDecks: PropTypes.func.isRequired,
  }

  static defaultProps = {}

  constructor(props) {
    super(props)
    this.handleRemoveDeck = this.handleRemoveDeck.bind(this)
  }

  handleRemoveDeck(deckId) {
    if (window.confirm('Are you sure you want to delete this deck?')) {
      this.props.removeDeck(deckId)
    }
  }

  render() {
    return (
      <TransitionMotion
        willEnter={() => ({ opacity: 0, left: -100 })}
        willLeave={() => ({ opacity: spring(0), left: spring(-100) })}
        styles={this.props.decks.map(d => ({
          key: `deck_${d.id}`,
          data: { deck: d },
          style: {
            opacity: spring(1),
            left: spring(0),
          },
        }))}
      >
        {interpolatedStyles => (
          <section className={`${styles['deck-grid']} wrap-row`}>
            {interpolatedStyles.map(config => (
              <Deck
                key={config.key}
                deck={config.data.deck}
                handleOnClick={() =>
                  this.props.push(`/view/${config.data.deck.id}`)
                }
                push={this.props.push}
                style={{
                  ...config.style,
                  display: 'relative',
                }}
              />
            ))}
            {interpolatedStyles.length === 0 && (
              /*
              <p key="no_value" className="center">
                Click the + button on the top left to add a deck
              </p>
              */
              <Splash loadSampleDecks={this.props.loadSampleDecks} />
            )}
          </section>
        )}
      </TransitionMotion>
    )
  }
}

export default DeckGrid
