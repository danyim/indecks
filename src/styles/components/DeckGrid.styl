@import '../_variables';

.deck-grid
  width: calc(100% - 5rem * 2)
  max-width: pageWidth
  margin: inherit 5rem 2rem 5rem
  flex-grow: 1
  flex-shrink: 0
  flex-flow: row wrap
  justify-content: center
  .grid-figure
    // This would be nice to have if it didn't bleed into the single cards
    transition: all .1s ease-out
    &:hover
      transform: scale(1.03, 1.03)
