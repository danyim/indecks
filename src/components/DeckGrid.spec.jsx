import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import DeckGrid from './DeckGrid'

const defaultProps = {
  decks: [{
    id: 'ABCDEFG',
    title: 'Deck title',
    description: 'Deck description',
    cards: []
  }],
  removeDeck: jest.fn()
}

function setup (props = defaultProps) {
  const wrapper = shallow(<DeckGrid {...props} />)

  return {
    props,
    wrapper
  }
}

function setupFull (props = defaultProps) {
  const wrapper = mount(<DeckGrid {...props} />)

  return {
    props,
    wrapper
  }
}

describe('DeckGrid', () => {
  it('should render self and subcomponents', () => {
    const tree = renderer.create(
      <DeckGrid {...defaultProps} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // This method is implemented but not used anywhere
  it.skip('should call the removeDeck prop when the "Delete" deck overlay button is clicked', () => {
    const handler = jest.fn()
    spyOn(window, 'confirm').and.returnValue(true)
    const { wrapper } = setup({
      ...defaultProps,
      removeDeck: handler
    })

    wrapper.find('Overlay button[children="Delete"]').first().simulate('click')
    expect(handler.mock.calls.length).toBe(1)
  })

  it('should display the correct amount of decks', () => {
    const { props, wrapper } = setupFull()

    expect(wrapper.find('Deck').length).toBe(props.decks.length)
  })

  it('should display a message if no decks are found', () => {
    const { wrapper } = setupFull({
      ...defaultProps,
      decks: []
    })
    expect(wrapper.find('p.center').exists()).toBe(true)
  })
})
