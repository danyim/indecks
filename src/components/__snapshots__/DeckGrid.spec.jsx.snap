// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckGrid should render self and subcomponents 1`] = `
<TransitionMotion
  didLeave={[Function]}
  styles={
    Array [
      Object {
        "data": Object {
          "deck": Object {
            "cards": Array [],
            "description": "Deck description",
            "id": "ABCDEFG",
            "title": "Deck title",
          },
        },
        "key": "deck_ABCDEFG",
        "style": Object {
          "left": Object {
            "damping": 26,
            "precision": 0.01,
            "stiffness": 170,
            "val": 0,
          },
          "opacity": Object {
            "damping": 26,
            "precision": 0.01,
            "stiffness": 170,
            "val": 1,
          },
        },
      },
    ]
  }
  willEnter={[Function]}
  willLeave={[Function]}
>
  <Component />
</TransitionMotion>
`

exports[`DeckGrid should render self and subcomponents 2`] = `
<TransitionMotion
  didLeave={[Function]}
  styles={Array []}
  willEnter={[Function]}
  willLeave={[Function]}
>
  <Component />
</TransitionMotion>
`
